app/src/main/java/com/duckduckgo/app/onboarding/ui/page/WelcomePage.kt

```diff
@@ -95,17 +94,23 @@ class WelcomePage : OnboardingPageFragment(R.layout.content_onboarding_welcome)
                 .flowOn(dispatcherProvider.io())
                 .collect(::render)
         }
+
+        requestNotificationsPermissions()
     }
 
-    @SuppressLint("InlinedApi")
     private fun requestNotificationsPermissions() {
         if (appBuildConfig.sdkInt >= android.os.Build.VERSION_CODES.TIRAMISU) {
-            requestPermission.launch(Manifest.permission.POST_NOTIFICATIONS)
+            event(WelcomePageView.Event.OnNotificationPermissionsRequested)
```

Comment:
I don't follow why we have this new event? Can't we just check here if the variant is assigned then request permissions otherwise do the welcome animation?

It looks like this new event throws more events to basically end up doing the same thing that we already have here?

Reply:
We need to create an Event to check if the experiment is enabled and the user variant on the ViewModel.
The welcome page has a different pattern which we don't use in other fragments, but I prefer to follow it for consistency.